packages/react-reconciler/src/ReactTypeOfMode.js

```diff
@@ -9,11 +9,11 @@
 
 export type TypeOfMode = number;
 
-export const NoMode = 0b00000;
-export const StrictMode = 0b00001;
-// TODO: Remove BlockingMode and ConcurrentMode by reading from the root
-// tag instead
-export const BlockingMode = 0b00010;
-export const ConcurrentMode = 0b00100;
-export const ProfileMode = 0b01000;
-export const DebugTracingMode = 0b10000;
+export const NoMode = /*           */ 0b000000;
+// TODO: Remove BlockingMode and ConcurrentMode by reading from the root tag instead
+export const BlockingMode = /*     */ 0b000001;
+export const ConcurrentMode = /*   */ 0b000010;
+export const ProfileMode = /*      */ 0b000100;
+export const DebugTracingMode = /* */ 0b001000;
+export const StrictModeL1 = /*     */ 0b010000;
+export const StrictModeL2 = /*     */ 0b100000;
```

Comment:
Another way to do this is to keep each category (render phase, effects, etc) as a separate bit

but then in the <StrictMode /> implementation, assign the correct bits for each level:

```js
if (level >= 1) {
  mode |= StrictRender;
}
if (level >= 2) {
  mode |= StrictEffects;
}
```

That way there's a single place where can decide which category of warnings goes into which level.

Reply:
Yeah, considered something along those lines but it seemed like it would balloon into a lot of flags (at least for level 1).

Are you just sharing this as an idea for consideration or do you think it's objectively better?

Reply:
I think effectively we're going to have one mode bit per level anyway, at least for what we have planned (just strict render and strict effects, right). So this does feel better to me.

Re: ballooning, yeah at some point we'll have to deal with running out of bits. But we have 32 so that will probably take a while.

Reply:
Maybe put another way: Seems unlikely that we're going to go back and _change_ what base strict mode means, so not much value in extracting the legacy strict mode things. Maybe future things (l2+)?

Reply:
So that case... maybe `StrictRender` becomes `StrictLegacy` and L2 is `StrictEffects` and L3 is `StrictWhateverElse`?

Reply:
> I think effectively we're going to have one mode bit per level anyway, at least for what we have planned (just strict render and strict effects, right). So this does feel better to me.

Ah. I think I misunderstood your suggestion for L1 maybe. I thought you were suggesting:
```js
if (levelProp >= 1) {
  mode |= StrictRender;
  mode |= DoubleLifecycles;
  mode |= NoLegacyContext;
  // etc.
}
if (levelProp >= 2) {
  mode |= StrictEffects;
}
```

Seems like you're more just suggesting a mode-name change?

"StrictRender" is a little weird still since it controls more than just that. That's kind of why I like `StrictLegacy` or something.

Reply:
> "StrictRender" is a little weird still since it controls more than just that.

My thinking was that it's designed to help discover render phase mutations/effects.

But yeah it's fine to group them like that. My other point was just that level prop of 2 turns on both mode bits.

Reply:
> But yeah it's fine to group them like that. My other point was just that level prop of 2 turns on both mode bits.

Oh yeah, for sure this is the case regardless of naming. 👍 

Was just planning on adding that new prop in a follow up PR.

Reply:
Okay. Renamed like we talked about.

I'll do the level prop follow up next.